packages/office-ui-fabric-react/src/components/Coachmark/Coachmark.tsx

```diff
@@ -0,0 +1,300 @@
+
+  constructor(props: ICoachmarkTypes) {
+    super(props);
+
+    // Set defaults for state
+    this.state = {
+      collapsed: props.collapsed!,
+      isBeaconAnimating: true,
+      isMeasuring: true,
+      entityInnerHostRect: {
+        width: 0,
+        height: 0
+      },
+      isMouseInProximity: false
+    };
+  }
+
+  public render(): JSX.Element {
+    let {
+      children,
+      beakWidth,
+      beakHeight,
+      target,
+      width,
+      height,
+      color,
+      beaconColorOne,
+      beaconColorTwo
+    } = this.props;
+
+    const classNames = getClassNames(getStyles, {
+      collapsed: this.state.collapsed,
+      isBeaconAnimating: this.state.isBeaconAnimating,
+      isMeasuring: this.state.isMeasuring,
+      entityHostHeight: this.state.entityInnerHostRect.height + 'px',
+      entityHostWidth: this.state.entityInnerHostRect.width + 'px',
+      width: width + 'px',
+      height: height + 'px',
+      color: color,
+      beaconColorOne: beaconColorOne,
+      beaconColorTwo: beaconColorTwo
+    });
+
+    return (
+      <PositioningContainer
+        target={ target }
+        offsetFromTarget={ beakHeight }
+        componentRef={ this._resolveRef('_positioningContainer') }
+      >
+        <div className={ classNames.root }>
+          <div className={ classNames.pulsingBeacon } />
+          <div
+            className={ classNames.translateAnimationContainer }
+            ref={ this._resolveRef('_translateAnimationContainer') }
+          >
+            <div className={ classNames.scaleAnimationLayer }>
+              <div className={ classNames.rotateAnimationLayer }>
+                {
+                  this._positioningContainer && <Beak
+                    width={ beakWidth }
+                    height={ beakHeight }
+                    left={ this.state.beakLeft }
+                    top={ this.state.beakTop }
+                  />
+                }
+                <FocusZone>
+                  <div
+                    className={ classNames.entityHost }
+                    ref={ this._resolveRef('_entityHost') }
+                    data-is-focusable={ true }
+                    onFocus={ this._onFocusHandler }
+                  >
+                    <div
+                      className={ classNames.entityInnerHost }
+                      ref={ this._resolveRef('_entityInnerHostElement') }
+                    >
+                      { children }
+                    </div>
+                  </div>
+                </FocusZone>
+              </div>
+            </div>
+          </div>
+        </div>
+      </PositioningContainer>
+    );
+  }
+
+  public componentWillReceiveProps(newProps: ICoachmarkTypes): void {
+    if (this.props.collapsed && !newProps.collapsed) {
+      // The coachmark is about to open
+      this._openCoachmark();
+    }
+  }
+
+  public componentDidMount(): void {
+    if (this.state.isMeasuring) {
+      this._async.setTimeout((): void => {
```

Comment:
use `this._async.requestAnimationFrame( ...) ` instead. It helps ensure that first dom changes occur, then when those are done all the measures can happen.